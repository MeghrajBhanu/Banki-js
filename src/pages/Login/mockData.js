export const mockLoginData = {
    name: 'mock user',
    email: 'mockuser@gmail.com',
    password: 'Qwety@123',
    pancard: 'RQTYI1234P',
  };
  export const mock_Invalid_LoginData = {
    name: 'm',
    email: 'mOCK1',
    password: 'ui',
    pancard: '1234',
  };