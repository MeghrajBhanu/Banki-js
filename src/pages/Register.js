import React, { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { register } from "../redux/actions/auth";
import { clearMessage } from "../redux/actions/message";
// import { AuthState } from '../redux/reducers/types';
import { useNavigate } from "react-router";
import { Link } from "react-router-dom";
import {
  validateEmail,
  validatePan,
  validatePass,
} from "../utils/InputValidations";
const Register = () => {
  const Navigate=useNavigate()
  const { message } = useSelector((state) => state.message);
  const dispatch = useDispatch();
  const [successful, setSuccessful] = useState(false);

  const [valuess, setValues] = useState({
    name: "",
    email: "",
    password: "",
    pancard: "",
  });
  const [disable, setDisable] = useState(true);
  const [isnameValid, setISNameValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isEmailValid, setISEmailValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isPanValid, setISPanValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isPassValid, setISPassValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });

  const handleInputValidation = (event) => {
    if (event.target.name === "name") {
      const { isInputValid } = validatePass(valuess.name);
      setISNameValid({
        isInputValid: isInputValid,
        errorMessage: "Invalid Name",
      });
    }
    if (event.target.name === "email") {
      const { isInputValid, errorMessage } = validateEmail(valuess.email);
      setISEmailValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (event.target.name === "pancard") {
      const { isInputValid, errorMessage } = validatePan(valuess.pancard);
      setISPanValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (event.target.name === "password") {
      const { isInputValid, errorMessage } = validatePass(valuess.password);
      setISPassValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (
      !isnameValid.isInputValid ||
      !isEmailValid.isInputValid ||
      !isPanValid.isInputValid ||
      !isPassValid.isInputValid
    ) {
      setDisable(true);
    } else {
      setDisable(false);
    }
  };
  const handleChange = (e) => {
    setValues({ ...valuess, [e.target.name]: e.target.value });
    handleInputValidation(e);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    console.log(valuess);
    dispatch(register(valuess))
      .then(() => {
        setSuccessful(true);
        setValues({
          name: "",
          email: "",
          password: "",
          pancard: "",
        });
      })
      .catch(() => {
        setSuccessful(false);
      });
    setDisable(true);
    setTimeout(() => {
      dispatch(clearMessage());
    }, 9000);
  };
  return (
    <section className="vh-100" style={{ backgroundColor: "#eee" }}>
      <div className="container h-100">
        <div className="row d-flex justify-content-center align-items-center h-100">
          <div className="col-lg-12 col-xl-11">
            <div className="card text-black" style={{ borderRadius: "25px" }}>
              <div className="card-body p-md-5">
                <div className="row justify-content-center">
                  <div className="col-md-10 col-lg-6 col-xl-5 order-2 order-lg-1">
                    <p className="text-center h1 fw-bold mb-5 mx-1 mx-md-4 mt-4">
                      Register
                    </p>

                    <form className="mx-1 mx-md-4">
                      <div className="d-flex flex-row align-items-center mb-4">
                        <i className="fas fa-user fa-lg me-3 fa-fw "></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="text"
                            name="name"
                            value={valuess.name}
                            id="form3Example1c"
                            className="form-control"
                            placeholder="Your Name"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isnameValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ "margin-left": "50px" }}
                        >
                          {isnameValid.errorMessage}
                        </div>
                      ) : null}
                      <div className="d-flex flex-row align-items-center mb-3">
                        <i className="fas fa-envelope fa-lg me-3 fa-fw"></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="email"
                            name="email"
                            value={valuess.email}
                            id="form3Example3c"
                            className="form-control"
                            placeholder="Your Email"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isEmailValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ "margin-left": "50px" }}
                        >
                          {isEmailValid.errorMessage}
                        </div>
                      ) : null}
                      <div className="d-flex flex-row align-items-center mb-4">
                        <i className="fas fa-solid fa-id-card fa-lg me-3 fa-fw "></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="text"
                            name="pancard"
                            value={valuess.pancard}
                            id="form3Example4cd"
                            className="form-control"
                            placeholder="Pancard"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isPanValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ "margin-left": "50px" }}
                        >
                          {isPanValid.errorMessage}
                        </div>
                      ) : null}

                      <div className="d-flex flex-row align-items-center mb-4">
                        <i className="fas fa-key fa-lg me-3 fa-fw"></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="password"
                            name="password"
                            id="form3Example4c"
                            className="form-control"
                            placeholder="Password"
                            value={valuess.password}
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isPassValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ "margin-left": "50px" }}
                        >
                          {isPassValid.errorMessage}
                        </div>
                      ) : null}

                      <div className="form-check d-flex justify-content-center mb-5">
                        <input
                          className="form-check-input me-2"
                          type="checkbox"
                          value=""
                          id="form2Example3c"
                        />
                        <label
                          className="form-check-label"
                          htmlFor="form2Example3"
                        >
                          I agree all statements in{" "}
                          <a href="#!">Terms of service</a>
                        </label>
                      </div>

                      <div className="d-flex justify-content-center mx-4 mb-3 mb-lg-4">
                        <button
                          type="submit"
                          className="btn btn-primary btn-lg"
                          onClick={onSubmit}
                          disabled={disable}
                        >
                          Register
                        </button>
                      </div>
                    </form>
                    {message && (
                      <div className="form-group">
                        <div
                          className={
                            successful
                              ? "alert alert-success"
                              : "alert alert-danger"
                          }
                          role="alert"
                        >
                          {message === "registartion done" ? (
                            <p>
                              Account has been registered
                              <br />
                              Please <Link to="/login">Login</Link> to continue

                            </p>
                          ) : (
                            message
                          )}
                        </div>
                      </div>
                    )}
                  </div>
                  <div className="col-md-10 col-lg-6 col-xl-7 d-flex align-items-center order-1 order-lg-2">
                    <img
                      src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-registration/draw1.webp"
                      className="img-fluid"
                      alt="a animated character walking"
                    ></img>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};
export default Register;
