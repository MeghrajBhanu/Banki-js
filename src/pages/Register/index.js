import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { register } from "../../redux/actions/auth";
import { clearMessage } from "../../redux/actions/message";
import { Link } from "react-router-dom";
import {
  validateEmail,
  validateName,
  validatePan,
  validatePass,
} from "../../utils/InputValidations";

/**
 * Registartion component handles users registration
 * @returns {React.ReactElement}
 */
const Register = () => {
  /**
   * 1.Store and dispatcher
   **/
  const { message } = useSelector((state) => state.message);
  const dispatch = useDispatch();

  /**
   * 2.Component State
   **/
  const [successful, setSuccessful] = useState(false);
  const [valuess, setValues] = useState({
    name: "",
    email: "",
    password: "",
    pancard: "",
  });
  const [disable, setDisable] = useState(true);
  const [isnameValid, setISNameValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isEmailValid, setISEmailValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isPanValid, setISPanValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });
  const [isPassValid, setISPassValid] = useState({
    isInputValid: false,
    errorMessage: "",
  });

  /**
   * 3.functions for handling inputs and form submission
   **/
  const handleInputValidation = (event) => {
    if (event.target.name === "name") {
      const { isInputValid, errorMessage } = validateName(event.target.value);
      setISNameValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (event.target.name === "email") {
      const { isInputValid, errorMessage } = validateEmail(event.target.value);
      setISEmailValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (event.target.name === "pancard") {
      const { isInputValid, errorMessage } = validatePan(event.target.value);
      setISPanValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (event.target.name === "password") {
      const { isInputValid, errorMessage } = validatePass(event.target.value);
      setISPassValid({
        isInputValid: isInputValid,
        errorMessage: errorMessage,
      });
    }
    if (
      !isnameValid.isInputValid ||
      !isEmailValid.isInputValid ||
      !isPanValid.isInputValid ||
      !isPassValid.isInputValid
    ) {
      setDisable(true);
    } else {
      setDisable(false);
    }
  };
  const handleChange = (e) => {
    setValues({ ...valuess, [e.target.name]: e.target.value });
    handleInputValidation(e);
  };

  const onSubmit = (e) => {
    e.preventDefault();

    console.log(valuess);
    dispatch(register(valuess))
      .then(() => {
        setSuccessful(true);
        setValues({
          name: "",
          email: "",
          password: "",
          pancard: "",
        });
      })
      .catch(() => {
        setSuccessful(false);
      });
    setDisable(true);
    setTimeout(() => {
      dispatch(clearMessage());
    }, 9000);
  };

  /**
   * 4.Page onload code cleanup
   **/
  useEffect(() => {
    dispatch(clearMessage());
  }, []);

  /**
   * 5.Actual content rendered on screen
   **/
  return (
    <section className="vh-100" style={{ backgroundColor: "#eee" }}>
      <div className="container h-100">
        <div className="row d-flex justify-content-center align-items-center h-100">
          <div className="col-lg-12 col-xl-11">
            <div className="card text-black" style={{ borderRadius: "25px" }}>
              <div className="card-body p-md-5">
                <div className="row justify-content-center">
                  <div className="col-md-10 col-lg-6 col-xl-5 order-2 order-lg-1">
                    <p className="text-center h1 fw-bold mb-5 mx-1 mx-md-4 mt-4">
                      Register
                    </p>

                    <form className="mx-1 mx-md-4">
                      <div className="d-flex flex-row align-items-center mb-4">
                        <i className="fas fa-user fa-lg me-3 fa-fw "></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="text"
                            aria-label="name"
                            name="name"
                            value={valuess.name}
                            id="form3Example1c"
                            className="form-control"
                            placeholder="Your Name"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isnameValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ marginLeft: "50px" }}
                        >
                          {isnameValid.errorMessage}
                        </div>
                      ) : null}
                      <div className="d-flex flex-row align-items-center mb-3">
                        <i className="fas fa-envelope fa-lg me-3 fa-fw"></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="email"
                            aria-label="email"
                            name="email"
                            value={valuess.email}
                            id="form3Example3c"
                            className="form-control"
                            placeholder="Your Email"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isEmailValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ marginLeft: "50px" }}
                        >
                          {isEmailValid.errorMessage}
                        </div>
                      ) : null}
                      <div className="d-flex flex-row align-items-center mb-4">
                        <i className="fas fa-solid fa-id-card fa-lg me-3 fa-fw "></i>
                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="text"
                            aria-label="pancard"
                            name="pancard"
                            value={valuess.pancard}
                            id="form3Example4cd"
                            className="form-control"
                            placeholder="Pancard"
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isPanValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ marginLeft: "50px" }}
                        >
                          {isPanValid.errorMessage}
                        </div>
                      ) : null}

                      <div className="d-flex flex-row align-items-center mb-4">
                        <i
                          className="fas fa-key fa-lg me-3 fa-fw "
                          data-coreui-toggle="tooltip"
                          data-coreui-html="true"
                          title="length: 8-15 characters ,
                                lowercase letter : atleat one, 
                                uppercase letter : atleat one,
                                numeric digit :atleat one,
                                special character : atleat one"
                        ></i>

                        <div className="form-outline flex-fill mb-0">
                          <input
                            type="password"
                            aria-label="password"
                            name="password"
                            id="form3Example4c"
                            className="form-control"
                            placeholder="Password"
                            value={valuess.password}
                            onChange={handleChange}
                            onBlur={handleInputValidation}
                            required="required"
                          />
                        </div>
                      </div>
                      {!isPassValid.isInputValid ? (
                        <div
                          className="text-danger mb-3"
                          style={{ marginLeft: "50px" }}
                        >
                          {isPassValid.errorMessage}
                        </div>
                      ) : null}

                      <div className="d-flex justify-content-center mx-4 mb-3 mb-lg-4">
                        <button
                          name="Register"
                          aria-label="Register"
                          type="submit"
                          className="btn btn-primary btn-lg"
                          onClick={onSubmit}
                          disabled={disable}
                        >
                          Register
                        </button>
                      </div>
                    </form>
                    {message && (
                      <div className="form-group">
                        <div
                          className={
                            successful
                              ? "alert alert-success"
                              : "alert alert-danger"
                          }
                          role="alert"
                        >
                          {message === "registartion done" ? (
                            <p>
                              Account has been registered
                              <br />
                              Please <Link to="/login">Login</Link> to continue
                            </p>
                          ) : (
                            message
                          )}
                        </div>
                      </div>
                    )}
                  </div>
                  <div className="col-md-10 col-lg-6 col-xl-7 d-flex align-items-center order-1 order-lg-2">
                    <img
                      src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-registration/draw1.webp"
                      className="img-fluid"
                      alt="a animated character walking"
                    ></img>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};
export default Register;
