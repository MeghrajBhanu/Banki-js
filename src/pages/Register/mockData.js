export const mockRegisterData = {
    name: 'mock user',
    email: 'mockuser@gmail.com',
    password: 'Qwety@123',
    confirmpassword: 'Qwety@123',
    pancard: 'RQTYI1234P',
  };
  export const mock_Invalid_RegisterData = {
    name: 'm',
    email: 'mOCK1',
    password: 'ui',
    confirmpassword: 'Qwety@123',
    pancard: '1234',
  };