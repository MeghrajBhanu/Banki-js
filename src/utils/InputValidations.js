export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}