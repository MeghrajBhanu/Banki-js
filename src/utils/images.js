import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;