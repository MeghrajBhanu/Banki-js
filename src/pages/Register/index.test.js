import { render, screen } from "@testing-library/react";
import { renderComponent } from "../../utils/test_helper";
import Register from ".";
import userEvent from "@testing-library/user-event";
import { mockRegisterData, mock_Invalid_RegisterData } from "./mockData";
const user = userEvent.setup();
describe("Register Page", () => {
  test("should show register heading", () => {
    render(renderComponent(<Register />));
    const value = screen.getAllByText(/Register/i);
    expect(value[0]).toBeInTheDocument();
  });

  test("should render the initial form", () => {
    render(renderComponent(<Register />));
    expect(screen.getByRole("textbox", { name: "name" })).toBeInTheDocument();
    expect(screen.getByRole("textbox", { name: "email" })).toBeInTheDocument();
    expect(
      screen.getByRole("textbox", { name: "pancard" })
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: "Register" })
    ).toBeInTheDocument();
  });
  test("should handle input changes", async () => {
    render(renderComponent(<Register />));
    const name = screen.getByRole("textbox", { name: "name" });
    const email = screen.getByRole("textbox", { name: "email" });
    const password = screen.getByPlaceholderText("Password");
    const pancard = screen.getByRole("textbox", { name: "pancard" });
    await user.type(name, mockRegisterData.name);
    expect(name).toHaveValue(mockRegisterData.name);

    await user.type(email, mockRegisterData.email);
    expect(email).toHaveValue(mockRegisterData.email);

    await user.type(password, mockRegisterData.password);
    expect(password).toHaveAttribute("type", "password");
    expect(password).toHaveValue(mockRegisterData.password);

    await user.type(pancard, mockRegisterData.pancard);
    expect(pancard).toHaveValue(mockRegisterData.pancard);
  });
  test("should handle errors in input", async () => {
    render(renderComponent(<Register />));
    const name = screen.getByRole("textbox", { name: "name" });
    const email = screen.getByRole("textbox", { name: "email" });
    const password = screen.getByPlaceholderText("Password");
    const pancard = screen.getByRole("textbox", { name: "pancard" });

    name.focus();
    await user.tab();
    expect(screen.getByText("Name is required")).toBeInTheDocument();

    email.focus();
    await user.tab();
    expect(screen.getByText("Email Address is required")).toBeInTheDocument();

    password.focus();
    await user.tab();
    expect(screen.getByText("Password Required")).toBeInTheDocument();

    pancard.focus();
    await user.tab();
    expect(screen.getByText("Pan Number is required")).toBeInTheDocument();

    await user.type(name, mock_Invalid_RegisterData.name);
    expect(screen.getByText("Invalid Name")).toBeInTheDocument();

    await user.type(email, mock_Invalid_RegisterData.email);
    expect(screen.getByText("Invalid Email Address")).toBeInTheDocument();

    await user.type(password, mock_Invalid_RegisterData.password);
    expect(screen.getByText("Invalid Password")).toBeInTheDocument();

    await user.type(pancard, mock_Invalid_RegisterData.pancard);
    expect(screen.getByText("Invalid Pan Number")).toBeInTheDocument();
  });
});
