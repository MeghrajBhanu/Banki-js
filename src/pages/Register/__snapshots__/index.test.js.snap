// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Page snapshot test 1`] = `
<section
  className="vh-100 mt-1"
  style={
    Object {
      "backgroundColor": "#eee",
    }
  }
>
  <div
    className="container h-100"
  >
    <div
      className="row d-flex justify-content-center align-items-center h-100"
    >
      <div
        className="col-lg-12 col-xl-11"
        style={
          Object {
            "marginBottom": "50px",
          }
        }
      >
        <div
          className="card text-black"
          style={
            Object {
              "borderRadius": "25px",
            }
          }
        >
          <div
            className="card-body "
          >
            <div
              className="row justify-content-center"
            >
              <div
                className="col-md-10 col-lg-6 col-xl-5 order-2 order-lg-1"
              >
                <p
                  className="text-center h1 fw-bold mb-5 mx-1 mx-md-4 mt-4"
                >
                  Register
                </p>
                <form
                  className="mx-1 mx-md-4"
                >
                  <div
                    className="d-flex flex-row align-items-center mb-4"
                  >
                    <i
                      className="fas fa-user fa-lg me-3 fa-fw "
                    />
                    <div
                      className="form-outline flex-fill mb-0"
                    >
                      <input
                        aria-label="name"
                        className="form-control"
                        id="form3Example1c"
                        name="name"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Your Name"
                        required="required"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="text-danger mb-3"
                    style={
                      Object {
                        "marginLeft": "50px",
                      }
                    }
                  />
                  <div
                    className="d-flex flex-row align-items-center mb-3"
                  >
                    <i
                      className="fas fa-envelope fa-lg me-3 fa-fw"
                    />
                    <div
                      className="form-outline flex-fill mb-0"
                    >
                      <input
                        aria-label="email"
                        className="form-control"
                        id="form3Example3c"
                        name="email"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Your Email"
                        required="required"
                        type="email"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="text-danger mb-3"
                    style={
                      Object {
                        "marginLeft": "50px",
                      }
                    }
                  />
                  <div
                    className="d-flex flex-row align-items-center mb-4"
                  >
                    <i
                      className="fas fa-solid fa-id-card fa-lg me-3 fa-fw "
                    />
                    <div
                      className="form-outline flex-fill mb-0"
                    >
                      <input
                        aria-label="pancard"
                        className="form-control"
                        id="form3Example4cd"
                        name="pancard"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Pancard"
                        required="required"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="text-danger mb-3"
                    style={
                      Object {
                        "marginLeft": "50px",
                      }
                    }
                  />
                  <div
                    className="d-flex flex-row align-items-center mb-4"
                  >
                    <i
                      className="fas fa-key fa-lg me-3 fa-fw "
                      data-coreui-html="true"
                      data-coreui-toggle="tooltip"
                      title="length: 8-15 characters ,
 lowercase letter : atleat one, 
 uppercase letter : atleat one,
 numeric digit :atleat one,
 special character : atleat one"
                    />
                    <div
                      className="form-outline flex-fill mb-0"
                    >
                      <input
                        aria-label="password"
                        className="form-control"
                        id="form3Example4c"
                        name="password"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Password"
                        required="required"
                        title="Must contain at least one number,one special character and one uppercase and lowercase letter, and at least 8 or more characters"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="text-danger mb-3"
                    style={
                      Object {
                        "marginLeft": "50px",
                      }
                    }
                  />
                  <div
                    className="d-flex flex-row align-items-center mb-4"
                  >
                    <i
                      className="fas fa-lock fa-lg me-3 fa-fw "
                      data-coreui-html="true"
                      data-coreui-toggle="tooltip"
                      title="length: 8-15 characters ,
 lowercase letter : atleat one, 
 uppercase letter : atleat one,
 numeric digit :atleat one,
 special character : atleat one"
                    />
                    <div
                      className="form-outline flex-fill mb-0"
                    >
                      <input
                        aria-label="confirmpassword"
                        className="form-control"
                        id="form3Example4c"
                        name="confirmpassword"
                        onBlur={[Function]}
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        required="required"
                        title="Must contain at least one number,one special character and one uppercase and lowercase letter, and at least 8 or more characters"
                        type="password"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="text-danger mb-3"
                    style={
                      Object {
                        "marginLeft": "50px",
                      }
                    }
                  />
                  <div
                    className="d-flex justify-content-center mx-4 mb-3 mb-lg-4"
                  >
                    <button
                      aria-label="Register"
                      className="btn btn-primary btn-lg"
                      disabled={true}
                      name="Register"
                      onClick={[Function]}
                      type="submit"
                    >
                      Register
                    </button>
                  </div>
                </form>
                <div
                  className="Toastify"
                />
              </div>
              <div
                className="col-md-10 col-lg-6 col-xl-7 d-flex align-items-center order-1 order-lg-2"
              >
                <img
                  alt="a animated character walking"
                  className="img-fluid"
                  src="https://mdbcdn.b-cdn.net/img/Photos/new-templates/bootstrap-registration/draw1.webp"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
